### PR TITLE
Add support for specifying visibility for Posit Cloud and shinyapps.io deploys

### PR DESCRIPTION
**Pull request by mslynch**
Resolves https://github.com/rstudio/rsconnect-python/issues/356.

## Intent

This change adds support for specifying a `--visibility` flag to shinyapps.io deploys. Visibility can be `public` or `private`.

## Type of Change

- [ ] Bug Fix            
- [x] New Feature       
- [ ] Breaking Change   

## Approach

If the deploy creates a new application and `--visibility` is provided, the visibility is set. If we're updating an existing application and `--visibility` is provided, we get the existing visibility, compare it to the provided visibility, and update it if the values differ.

## Automated Tests

Updated the automated tests to pass `--visibility` and added/updated HTTP mocks to match the change in API calls.

## Directions for Reviewers

Important cases to test are:
- new deploy on shinyapps.io with `--visibility`
- redeploy on shinyapps.io with `--visibility`

## Checklist

- [x] I have updated [CHANGELOG.md](../CHANGELOG.md) to cover notable changes.
- [x] I have updated all related GitHub issues to reflect their current state.

### CODE REVIEW

**Review by mbaynton — APPROVED** [review]

**Review by omar-rs — APPROVED** [review]

**Comment by mslynch** [comment]
@scubastevew Take a look at this change when you get a chance.

**Comment by scubastevew** [comment]
Tested: 
- [ ] deploying to shinyapps.io with no flag -> defaults to public
- [ ] deploying to shinyapps.io with with `visibility public` -> sets application to public
- [ ] deploying to shinyapps.io with with `visibility private` -> sets application to private
- [ ] updating a public application on shinyapps.io with with `visibility private` -> sets application to private
- [ ] updating a private application on shinyapps.io with with `visibility public` -> sets application to public

File: rsconnect/api.py [hunk 1]
```diff
@@ -1254,9 +1270,23 @@ def prepare_deploy(self, app_id: typing.Optional[int], app_name: str, bundle_siz
...
         else:
             application = self._rstudio_client.get_application(app_id)
-        self._server.handle_bad_response(application)
+            self._server.handle_bad_response(application)
+
+            if visibility is not None:
+                if visibility != application["deployment"]["properties"]["application.visibility"]:
+                    self._rstudio_client.update_application_property(
+                        application["id"], "application.visibility", visibility
+                    )
```
Comment (mmarchetti) [inline rsconnect/api.py:1288]: Does this need to call `handle_bad_response`?

Comment (mslynch) [reply rsconnect/api.py:1259]: Yes :) thanks for catching that.

**Pull request merged**